### PR TITLE
Fix potential nil pointer

### PR DESCRIPTION
**Pull request by cwdsuzhou**
**What type of PR is this?**

/kind bug

**What this PR does / why we need it**:

Fix potential nil pointer

**Which issue(s) this PR fixes**:

Fixes #

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note
NONE
```

### CODE REVIEW

File: pkg/controller/controller.go [hunk 1]
```diff
@@ -424,7 +424,8 @@ func (p *csiProvisioner) ProvisionExt(options controller.ProvisionOptions) (*v1.
 
 		switch options.PVC.Spec.DataSource.Kind {
 		case snapshotKind:
-			if *(options.PVC.Spec.DataSource.APIGroup) != snapshotAPIGroup {
+			if options.PVC.Spec.DataSource.APIGroup == nil ||
+				*(options.PVC.Spec.DataSource.APIGroup) != snapshotAPIGroup {
 				return nil, controller.ProvisioningFinished, fmt.Errorf("the PVC source does not belong to the right APIGroup. Expected %s, Got %s", snapshotAPIGroup, *(options.PVC.Spec.DataSource.APIGroup))
```
Comment (jsafrane) [inline pkg/controller/controller.go]: you may dereference nil here

**Comment by jsafrane** [comment]
Were you able to actually hit this nil dereference or are you just reading code? I am very interested in reproducer, because something must have gone terribly wrong in validation and it should be fixed also there.

**Comment by cwdsuzhou** [comment]
> Were you able to actually hit this nil dereference or are you just reading code? I am very interested in reproducer, because something must have gone terribly wrong in validation and it should be fixed also there.

Thanks, I found it has been validated at apiserver.  

I am just reading the codes

/close

**Pull request closed without merging**